creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -1004,6 +927,201 @@ class SnapbackSM {
+   *    and record replica set updates to issue for any unhealthy replicas
+   *
+   * Purpose for the if/else case is that if the current node is a primary, issue reconfig or sync requests.
+   * Else, if the current node is a secondary, only issue reconfig requests.
+   *
+   * @notice this will issue sync to healthy secondary and update replica set away from unhealthy secondary
+   */
+  async aggregateReconfigAndPotentialSyncOps (nodeUsers, unhealthyPeers) {
+    let requiredUpdateReplicaSetOps = []
+    let potentialSyncRequests = []
+
+    for (const nodeUser of nodeUsers) {
+      const { primarySpID, secondary1SpID, secondary2SpID } = nodeUser
+
+      // If these spIds are undefined, it means that the Discovery Node does not have the latest code to return
+      // these fields. Fallback to the original aggregation logic. If present, use the spIds to determine whether
+      // reconfigs/syncs are necessary.
+      if (primarySpID && secondary1SpID && secondary2SpID) {
+        await this._aggregateOpsWithQueriedSpIds({ nodeUser, unhealthyPeers, requiredUpdateReplicaSetOps, potentialSyncRequests })
+      } else {
+        // TODO: remove this else case once all the Discovery Nodes have upgraded and return the necessary replica set spIds
+        await this._aggregateOpsWithoutQueriedSpIds({ nodeUser, unhealthyPeers, requiredUpdateReplicaSetOps, potentialSyncRequests })
+      }
+    }
+
+    return { requiredUpdateReplicaSetOps, potentialSyncRequests }
+  }
+
+  /**
+   * Used to determine the `requiredUpdateReplicaSetOps` and `potentialSyncRequests` for a given nodeUser.
+   * @param {Object} nodeUser { primary, secondary1, secondary2, primarySpID, secondary1SpID, secondary2SpID, user_id, wallet}
+   * @param {Set<string>} unhealthyPeers set of unhealthy peers
+   * @param {Object[]} requiredUpdateReplicaSetOps array of {...nodeUsers, unhealthyReplicas: {string[]} endpoints of unhealthy rset nodes }
+   * @param {Object[]} potentialSyncRequests array of {...nodeUsers, endpoint: {string} endpoint to sync to }
+   */
+  async _aggregateOpsWithQueriedSpIds ({ nodeUser, unhealthyPeers, requiredUpdateReplicaSetOps, potentialSyncRequests }) {
+    const { primary, secondary1, secondary2, primarySpID, secondary1SpID, secondary2SpID } = nodeUser
+    let unhealthyReplicas = []
+
+    /**
+     * If this node is primary for user, check both secondaries for health
+     * Enqueue SyncRequests against healthy secondaries, and enqueue UpdateReplicaSetOps against unhealthy secondaries
+     */
+    let replicaSetNodesToObserve = [
+      { endpoint: secondary1, spId: secondary1SpID },
+      { endpoint: secondary2, spId: secondary2SpID }
+    ]
+
+    if (primary === this.endpoint) {
+      // filter out false-y values to account for incomplete replica sets
+      const secondariesInfo = replicaSetNodesToObserve.filter(entry => entry.endpoint && entry.spId)
```

Comment:
spID check is still here and below on line 1016

Reply:
updating